cdc/owner.go

```diff
@@ -750,6 +754,9 @@ func (o *Owner) flushChangeFeedInfos(ctx context.Context) error {
 		}
 	}
 	if time.Since(o.gcSafepointLastUpdate) > GCSafepointUpdateInterval {
+		if (time.Now().Unix() - int64(minCheckpointTs)) > o.gcBLT {
+			minCheckpointTs = uint64(time.Now().Unix() - o.gcBLT)
+		}
 		actual, err := o.pdClient.UpdateServiceGCSafePoint(ctx, CDCServiceSafePointID, o.gcTTL, minCheckpointTs)
```

Comment:
Please add comments about how the `service gc safepoint` is decided in every single scenario, and add unit tests for each of these scenarios

Reply:
ok~